docs/dsl_quick_start.md

```diff
@@ -0,0 +1,176 @@
+# DSL Quick Start
+
+The MSCCL++ DSL (Domain Specific Language) provides a high-level Python API for defining custom collective communication algorithms. This guide will help you get started with writing and testing your own communication patterns.
+
+## Installation
+
+You can follow the same steps in the installation [Quick Start](quickstart). 
+
+After finishing the installation in the quick start section, you can add the following steps to install some default algorithms from the DSL:
+
+```bash
+python3 -m mscclpp --install
+```
+
+## Your First Algorithm: AllGather
+
+Let's walk through a simple AllGather algorithm to understand the DSL basics. This example demonstrates the key concepts without diving into all the advanced features.
+
+### Complete Example
+
+```python
+from mscclpp.language import *
+
+def simple_allgather(name):
+    """
+    A simple AllGather implementation using the MSCCL++ DSL.
+    
+    This example demonstrates a 2-GPU AllGather where each GPU sends
+    its data to all other GPUs, so all GPUs end up with everyone's data.
+    
+    Args:
+        name: Algorithm name for identification
+    """
+    num_gpus = 2
```

Comment:
Just feel the code maybe staled if we don't have a pipeline to verify it. Suggest to use Jupiter notebook for this. Then we can run the notebook in our pipeline. To make sure everything is OK